Formula/hcloud.rb

```diff
@@ -0,0 +1,28 @@
+class Hcloud < Formula
+  desc "Command-line interface for Hetzner Cloud"
+  homepage "https://github.com/hetznercloud/cli"
+  url "https://github.com/hetznercloud/cli/archive/v1.3.2.tar.gz"
+  sha256 "1cac1e7a5618abce0e6e2d5da40afc153fd88508557afdf90d0f5be54abec9a9"
+
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    (buildpath/"src/github.com/hetznercloud/cli").install buildpath.children
+
+    cd "src/github.com/hetznercloud/cli" do
+      ldflags = "-w -X github.com/hetznercloud/cli.Version=v#{version}"
+      system "go", "build", "-o", bin/"hcloud", "-ldflags", ldflags,
+                   "./cmd/hcloud"
+    end
+
+    output = Utils.popen_read("#{bin}/hcloud completion bash")
+    (bash_completion/"hcloud").write output
+    output = Utils.popen_read("#{bin}/hcloud completion zsh")
+    (zsh_completion/"_hcloud").write output
+  end
+
+  test do
+    assert_match "hcloud v#{version}", shell_output("#{bin}/hcloud version")
```

Comment:
Can we do a better test then checking the version?

Reply:
Almost every command requires an authentication token to be present, so I’m afraid this is all we can test.

Reply:
Testing a command that fails and checking the error would work.

Reply:
There is no command which is guaranteed to always return an error.

I don’t see why checking the version output is insufficient to verify the program was built and installed correctly. We should not test the program itself here. It may be a weak argument, but there are already formulae which just check the version output.

Reply:
```ruby
  test do
    (testpath/".config/hcloud/cli.toml").write <<~EOS
    active_context = "homebrew"
    
    [[contexts]]
      name = "homebrew"
      token = "test"
    EOS
    assert_match "homebrew", shell_output("#{bin}/hcloud context list")
  end
```